src/app/MessageDef.cpp

```diff
@@ -0,0 +1,2725 @@
+#endif // CHIP_DETAIL_LOGGING
+
+CHIP_ERROR LookForElementWithTag(const chip::TLV::TLVReader & aSrcReader, const uint64_t aTagInApiForm,
+                                  chip::TLV::TLVReader * apDstReader)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // make a copy of the Path reader
+    chip::TLV::TLVReader reader;
+    reader.Init(aSrcReader);
+
+    while (CHIP_NO_ERROR == (err = reader.Next()))
+    {
+        // Documentation says the result of GetType must be verified before calling GetTag
+        VerifyOrExit(chip::TLV::kTLVType_NotSpecified != reader.GetType(), err = CHIP_ERROR_INVALID_TLV_ELEMENT);
+
+        if (aTagInApiForm == reader.GetTag())
+        {
+            apDstReader->Init(reader);
+            break;
+        }
+    }
+
+exit:
+    ChipLogIfFalse((CHIP_NO_ERROR == err) || (CHIP_END_OF_TLV == err));
+
+    return err;
+}
+
+ParserBase::ParserBase() { }
+
+CHIP_ERROR ParserBase::GetReaderOnTag(const uint64_t aTagToFind, chip::TLV::TLVReader * const apReader) const
+{
+    return LookForElementWithTag(mReader, aTagToFind, apReader);
+}
+
+template <typename T>
+CHIP_ERROR ParserBase::GetUnsignedInteger(const uint8_t aContextTag, T * const apLValue) const
+{
+    return GetSimpleValue(aContextTag, chip::TLV::kTLVType_UnsignedInteger, apLValue);
+}
+
+template <typename T>
+CHIP_ERROR ParserBase::GetSimpleValue(const uint8_t aContextTag, const chip::TLV::TLVType aTLVType, T * const apLValue) const
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::TLV::TLVReader reader;
+
+    *apLValue = 0;
+
+    err = LookForElementWithTag(mReader, chip::TLV::ContextTag(aContextTag), &reader);
+    SuccessOrExit(err);
+
+    VerifyOrExit(aTLVType == reader.GetType(), err = CHIP_ERROR_WRONG_TLV_TYPE);
+
+    err = reader.Get(*apLValue);
+    SuccessOrExit(err);
+
+exit:
+    ChipLogIfFalse((CHIP_NO_ERROR == err) || (CHIP_END_OF_TLV == err));
+
+    return err;
+}
+
+ListParserBase::ListParserBase() { }
+
+CHIP_ERROR ListParserBase::Init(const chip::TLV::TLVReader & aReader)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // make a copy of the reader here
+    mReader.Init(aReader);
+
+    VerifyOrExit(chip::TLV::kTLVType_Array == mReader.GetType(), err = CHIP_ERROR_WRONG_TLV_TYPE);
+
+    // This is just a dummy, as we're not going to exit this container ever
+    chip::TLV::TLVType OuterContainerType;
+    err = mReader.EnterContainer(OuterContainerType);
+
+exit:
+    ChipLogFunctError(err);
+
+    return err;
+}
+
+CHIP_ERROR ListParserBase::InitIfPresent(const chip::TLV::TLVReader & aReader, const uint8_t aContextTagToFind)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::TLV::TLVReader reader;
+
+    err = LookForElementWithTag(aReader, chip::TLV::ContextTag(aContextTagToFind), &reader);
+    SuccessOrExit(err);
+
+    err = Init(reader);
+    SuccessOrExit(err);
+
+exit:
+    ChipLogIfFalse((CHIP_NO_ERROR == err) || (CHIP_END_OF_TLV == err));
+
+    return err;
+}
+
+CHIP_ERROR ListParserBase::Next(void)
```

Comment:
we should remove (void) and replace with (). We did this throughout CHIP using clang-tidy.

Reply:
Done, thanks